File: php/src/org/shypl/sna/impl/MmSocialNetwork.php
<?php

namespace org\shypl\sna\impl;

use org\shypl\sna\Adapter;
use org\shypl\sna\SocialNetwork;

final class MmSocialNetwork extends SocialNetwork {
	const ID = 2;

	/**
	 *
	 */
	public function __construct() {
		parent::__construct(self::ID, "mm");
	}

	/**
	 * @param array $parameters
	 *
	 * @return Adapter
	 */
	public function createAdapter(array $parameters) {
		return new MmAdapter($parameters);
	}
}

File: java/src/main/java/org/shypl/sna/impl/MmSocialNetwork.java
package org.shypl.sna.impl;

import org.shypl.common.util.DeclinationImpl;
import org.shypl.sna.Currency;
import org.shypl.sna.SocialNetwork;

public final class MmSocialNetwork extends SocialNetwork {
	public MmSocialNetwork() {
		super(2, "mm", "Мой Мир", new Currency("мэйлики", new DeclinationImpl("мэйлик", "мэйлика", "мэйликов")));
	}
}
